src/consensus/pbft/pbft.h

```diff
@@ -218,6 +309,19 @@ namespace pbft
       message_receiver_base->register_reply_handler(
         reply_handler_cb, client_proxy.get());
 
+      auto mark_stable_cb = [](void* ctx) {
```

Comment:
Why void*?

Reply:
I was doing this based on how did we did it in the PR https://github.com/microsoft/CCF/pull/405, I didn't realize this does not apply. I have updated the callback you use typed pointer. We can change these callbacks further to e.g. `std::functions` in a different PR.